client/my-sites/email/email-forwards-add/index.tsx

```diff
@@ -0,0 +1,121 @@
+import HeaderCake from 'calypso/components/header-cake';
+import Main from 'calypso/components/main';
+import SectionHeader from 'calypso/components/section-header';
+import PageViewTracker from 'calypso/lib/analytics/page-view-tracker';
+import EmailForwardingAddNewCompactList from 'calypso/my-sites/email/email-forwarding/email-forwarding-add-new-compact-list';
+import EmailHeader from 'calypso/my-sites/email/email-header';
+import {
+	emailManagement,
+	emailManagementAddEmailForwards,
+	emailManagementPurchaseNewEmailAccount,
+} from 'calypso/my-sites/email/paths';
+import getCurrentRoute from 'calypso/state/selectors/get-current-route';
+import {
+	hasLoadedSiteDomains,
+	isRequestingSiteDomains,
+} from 'calypso/state/sites/domains/selectors';
+import { getSelectedSite } from 'calypso/state/ui/selectors';
+
+import './style.scss';
+
+type EmailForwardsAddProps = {
+	selectedDomainName: string;
+	source?: string;
+};
+
+// eslint-disable-next-line @typescript-eslint/no-empty-function
+const noop = (): void => {};
+
+const EmailForwardsAdd = ( { selectedDomainName, source }: EmailForwardsAddProps ): JSX.Element => {
+	const currentRoute = useSelector( getCurrentRoute );
+	const selectedSite = useSelector( getSelectedSite );
+
+	const hasLoadedDomains = useSelector( ( state ) =>
+		hasLoadedSiteDomains( state, selectedSite?.ID ?? null )
+	);
+	const requestingSiteDomains = useSelector( ( state ) =>
+		Boolean( selectedSite && isRequestingSiteDomains( state, selectedSite.ID ) )
+	);
+	const areDomainsLoading = requestingSiteDomains || ! hasLoadedDomains;
+
+	const translate = useTranslate();
+
+	const goToEmail = useCallback( (): void => {
+		if ( ! selectedSite ) {
+			return;
+		}
+
+		if ( source === 'purchase' ) {
+			page(
+				emailManagementPurchaseNewEmailAccount(
+					selectedSite.slug,
+					selectedDomainName,
+					currentRoute
+				)
+			);
+			return;
+		}
+
+		page( emailManagement( selectedSite.slug, selectedDomainName, currentRoute ) );
+	}, [ currentRoute, selectedDomainName, selectedSite, source ] );
+
+	const onAddEmailForwardSuccess = useCallback( () => {
+		if ( ! selectedSite ) {
+			return;
+		}
+
+		page( emailManagement( selectedSite.slug, selectedDomainName, currentRoute ) );
+	}, [ currentRoute, selectedDomainName, selectedSite ] );
+
+	const analyticsPath = emailManagementAddEmailForwards( ':site', ':domain', currentRoute );
+
+	return (
+		<div className="email-forwards-add">
+			<PageViewTracker path={ analyticsPath } title="Email Management > Add Email Forwards" />
+
+			<QueryProductsList />
+
+			{ selectedSite && <QuerySiteDomains siteId={ selectedSite.ID } /> }
+
+			<Main wideLayout={ true }>
+				<DocumentHead title={ translate( 'Add New Email Forwards' ) } />
+
+				<EmailHeader currentRoute={ currentRoute } selectedSite={ selectedSite } />
+
+				<HeaderCake onClick={ goToEmail }>
+					{ translate( 'Email Forwarding' ) + ': ' + selectedDomainName }
```

Comment:
I know you've copy/pasted existing code but I wonder if that approach will work in RTL languages. Maybe we should put everything inside the string that is localized like this:

```javascript
{ translate( 'Email Forwarding: %(domain)s', ... ) }
```